libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -1959,6 +1959,48 @@ namespace webassembly {
          int32_t __lttf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
          int32_t __unordtf2(uint64_t, uint64_t, uint64_t, uint64_t) const;
 
+         // security group api
+         /**
+          * Propose new participants to the security group.
+          *
+          * @ingroup security-group
+          * @param packed_participants - the buffer containing the packed participants.
+          * 
+          * @return -1 if proposing a new security group was unsuccessful, otherwise returns 0.
+         */
+         int64_t propose_security_group_participants_add(span<const char> packed_participants);
+
+         /**
+          * Propose to remove participants from the security group.
+          *
+          * @ingroup security-group
+          * @param packed_participants - the buffer containing the packed participants.
+          * 
+          * @return -1 if proposing a new security group was unsuccessful, otherwise returns 0.
+         */
+         int64_t propose_security_group_participants_remove(span<const char> packed_participants);
+
+         /**
+          * Check if the specified accounts are all in the active security group.
+          *
+          * @ingroup security-group
+          * @param packed_participants - the buffer containing the packed participants.
+          * 
+          * @return Returns true if the specified accounts are all in the active security group. 
+         */
+         bool in_active_security_group(span<const char> packed_participants) const;
+
+         /**
+          * Gets the active security group
+          *
+          * @ingroup security-group
+          * @param[out] packed_security_group - the buffer containing the packed security_group.
+          *
+          * @return Returns the size required in the buffer (if the buffer is too small, nothing is written).
```

Comment:
The comment pattern hear follows the `get_kv_parameters_packed` and I believe it's clear enough. The `db_get_i64` comments is actually not a very good fit here because the implementation uses the length of the buffer to decide whether to write the data, not based on whether the buffer pointer is 0.

Reply:
db_get_i64 uses if the buffer size is 0, otherwise it copies over whatever it can fit of the data, which isn't that useful.  Since there are other intrinsics that follow the same logic, then keep it.  If the db_* methods were new, I'd have changed them since I don't know what the purpose of copying the start of the buffer, with no way of getting the rest of the buffer without still getting the start.